pkg/machinepool/defaults.go

```diff
@@ -0,0 +1,35 @@
+package machinepool
+
+import (
+	"context"
+	"fmt"
+
+	capiexp "sigs.k8s.io/cluster-api/exp/api/v1alpha3"
+
+	"github.com/giantswarm/azure-admission-controller/pkg/mutator"
+)
+
+// setDefaultSpecValues checks if some optional field is not set, and sets
+// default values defined by upstream Cluster API.
+func (m *CreateMutator) setDefaultSpecValues(ctx context.Context, machinePool *capiexp.MachinePool) []mutator.PatchOperation {
+	var patches []mutator.PatchOperation
+
+	defaultSpecReplicas := m.setDefaultReplicaValue(ctx, machinePool)
+	if defaultSpecReplicas != nil {
+		patches = append(patches, *defaultSpecReplicas)
+	}
+
+	return patches
+}
+
+// setDefaultReplicaValue checks if Spec.Replicas has been set, and if it is
+// not, it sets its value to 1.
+func (m *CreateMutator) setDefaultReplicaValue(ctx context.Context, machinePool *capiexp.MachinePool) *mutator.PatchOperation {
+	if machinePool.Spec.Replicas == nil {
+		const defaultReplicas = "1"
```

Comment:
We normally define the constants in the top of the file, but I'm not blocking. Your call.

Reply:
I basically needed a local var, which is not used anywhere else, but then changed it to `const` without too much thinking, as it does not change nor it should change.